junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/discovery/DiscoveryTests.java

```diff
@@ -105,6 +107,24 @@ public void discoverCompositeSpec() {
 		assertEquals(7, engineDescriptor.getDescendants().size(), "# resolved test descriptors");
 	}
 
+	@Test
+	public void discoverTestTemplateMethodByUniqueId() {
+		LauncherDiscoveryRequest spec = request().selectors(
+			selectUniqueId(uniqueIdForTestTemplateMethod(TestTemplateClass.class, "testTemplate()"))).build();
+
+		TestDescriptor engineDescriptor = discoverTests(spec);
+		assertEquals(2, engineDescriptor.getDescendants().size(), "# resolved test descriptors");
+	}
+
+	@Test
+	public void discoverTestTemplateMethodByMethodReference() {
+		LauncherDiscoveryRequest spec = request().selectors(
```

Comment:
Maybe discoverTestTemplateMethodByClassReferenceAndMethodName() would be a better name?